src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -634,7 +664,67 @@ - (XCTestExpectation *)announceProviderToDevice:(MTRDevice *)device
     return responseExpectation;
 }
 
-- (void)test001_ReceiveOTAQuery
+/**
+ * Helper that checks if the OTA provider locations match.
+ */
+- (bool)AreEqual:(MTROtaSoftwareUpdateRequestorClusterProviderLocation *)pl1
+    providerLocation:(MTROtaSoftwareUpdateRequestorClusterProviderLocation *)pl2
+{
+    return [pl1.providerNodeID isEqualToNumber:pl2.providerNodeID] && [pl1.endpoint isEqualToNumber:pl2.endpoint] &&
+        [pl1.fabricIndex isEqualToNumber:pl2.fabricIndex];
+}
+
+- (void)test001_UpdateDefaultOTAProviderAndVerify
+{
+    // Test to update the default OTA provider for a device and verify that it's updated correctly.
+
+    MTROtaSoftwareUpdateRequestorClusterProviderLocation * providerLocation =
+        [[MTROtaSoftwareUpdateRequestorClusterProviderLocation alloc] init];
+    providerLocation.providerNodeID = @(kOTAProviderNodeId);
+    providerLocation.endpoint = @(kOTAProviderEndpointId);
+    providerLocation.fabricIndex = @(kOTAProviderFabricIndex);
+
+    dispatch_queue_t queue = dispatch_get_main_queue();
+    XCTestExpectation * updateDefaultOTAProviderExpectation =
+        [self expectationWithDescription:@"UpdateDefaultOTAProvider verified"];
+
+    [sController
+            getBaseDevice:kDeviceId1
+                    queue:queue
+        completionHandler:^(MTRBaseDevice * _Nullable device, NSError * _Nullable error) {
+            XCTAssertEqual(error.code, 0);
+            XCTAssertNotNil(device);
+
+            MTRBaseClusterOtaSoftwareUpdateRequestor * cluster =
+                [[MTRBaseClusterOtaSoftwareUpdateRequestor alloc] initWithDevice:device endpointID:@(0) queue:queue];
+            NSArray * value = [NSArray arrayWithObject:providerLocation];
+            [cluster
+                writeAttributeDefaultOTAProvidersWithValue:value
+                                                completion:^(NSError * _Nullable error) {
+                                                    XCTAssertNil(error);
+                                                    MTRReadParams * params = [[MTRReadParams alloc] init];
+                                                    params.filterByFabric = true;
+                                                    [cluster
+                                                        readAttributeDefaultOTAProvidersWithParams:params
+                                                                                        completion:^(NSArray * _Nullable readValue,
+                                                                                            NSError * _Nullable err) {
+                                                                                            XCTAssertEqual(error.code, 0);
+                                                                                            XCTAssertNotNil(readValue);
+                                                                                            XCTAssertTrue(
```

Comment:
Shoudl we not assert that readValue has length 1?

Reply:
N/A as I removed this for now